contracts/abstract/JBPayoutRedemptionPaymentTerminal3_1_1.sol

```diff
@@ -0,0 +1,1718 @@
+      // Keep a reference to the fee amount that was paid.
+      uint256 _fee;
+
+      // Get a reference to the project owner, which will receive tokens from paying the platform fee.
+      address _projectOwner = projects.ownerOf(_projectId);
+
+      // Get the amount of discount that should be applied to any fees taken.
+      // If the fee is zero or if the fee is being used by an address that doesn't incur fees, set the discount to 100% for convenience.
+      uint256 _feeDiscount = fee == 0 || isFeelessAddress[msg.sender]
+        ? JBConstants.MAX_FEE_DISCOUNT
+        : _currentFeeDiscount(_projectId);
+
+      // Take a fee from the `_distributedAmount`, if needed.
+      _fee = _feeDiscount == JBConstants.MAX_FEE_DISCOUNT
+        ? 0
+        : _takeFeeFrom(
+          _projectId,
+          _fundingCycle.shouldHoldFees(),
+          _distributedAmount,
+          _projectOwner,
+          _feeDiscount
+        );
+
+      unchecked {
+        // The net amount is the withdrawn amount without the fee.
+        netDistributedAmount = _distributedAmount - _fee;
+      }
+
+      // Transfer any remaining balance to the beneficiary.
+      if (netDistributedAmount > 0)
+        _transferFrom(address(this), _beneficiary, netDistributedAmount);
+    }
+
+    emit UseAllowance(
+      _fundingCycle.configuration,
+      _fundingCycle.number,
+      _projectId,
+      _beneficiary,
+      _amount,
+      _distributedAmount,
+      netDistributedAmount,
+      _memo,
+      _metadata,
+      msg.sender
+    );
+  }
+
+  /**
+    @notice
+    Pays out splits for a project's funding cycle configuration.
+
+    @param _projectId The ID of the project for which payout splits are being distributed.
+    @param _domain The domain of the splits to distribute the payout between.
+    @param _group The group of the splits to distribute the payout between.
+    @param _amount The total amount being distributed, as a fixed point number with the same number of decimals as this terminal.
+    @param _feeDiscount The amount of discount to apply to the fee, out of the MAX_FEE.
+
+    @return leftoverAmount If the leftover amount if the splits don't add up to 100%.
+    @return feeEligibleDistributionAmount The total amount of distributions that are eligible to have fees taken from.
+  */
+  function _distributeToPayoutSplitsOf(
+    uint256 _projectId,
+    uint256 _domain,
+    uint256 _group,
+    uint256 _amount,
+    uint256 _feeDiscount
+  ) internal returns (uint256 leftoverAmount, uint256 feeEligibleDistributionAmount) {
+    // Set the leftover amount to the initial amount.
+    leftoverAmount = _amount;
+    // The total percentage available to split
+    uint256 leftoverPercentage = JBConstants.SPLITS_TOTAL_PERCENT;
+
+    // Get a reference to the project's payout splits.
+    JBSplit[] memory _splits = splitsStore.splitsOf(_projectId, _domain, _group);
+
+    // Transfer between all splits.
+    for (uint256 _i; _i < _splits.length; ) {
```

Comment:
same here too

Reply:
this is a stack too deep thing. we could do via-ir, worth it?